lib/appium.js

```diff
@@ -165,37 +175,47 @@ class AppiumDriver extends BaseDriver {
 
     // sessionOverride server flag check
     // this will need to be re-thought when we go to multiple session support
-    if (this.args.sessionOverride && !!this.sessions && _.keys(this.sessions).length > 0) {
-      log.info('Session override is on. Deleting other sessions.');
-      for (let id of _.keys(this.sessions)) {
-        log.info(`    Deleting session '${id}'`);
-        try {
-          await this.deleteSession(id);
-        } catch (ign) {
-          // the error has already been logged in AppiumDriver.deleteSession
-          // continue
+    if (this.args.sessionOverride) {
+      const sessionIdsToDelete = await sessionsListGuard.acquire(AppiumDriver.name, () => _.keys(this.sessions));
```

Comment:
we see a lot of repetition of naming the lock here. could create a simple helper function:

```js
async guardedSessions (fn) {
  await sessionsListGuard.acquire(AppiumDriver.name, () => {
    return fn(this.sessions);
  });
}
```

then we could change lines like this to:

```js
const sessionIdsToDelete = await this.guardedSessions(_.keys);
```